### PR TITLE
ddmd cleanup and fixes

### PR DESCRIPTION
**Pull request by MartinNowak**
- cleanup makefiles
- drop -O -inline by default (too slow)
- remove unused headers and magicport
- update travis-ci test

### CODE REVIEW

File: src/win32.mak [hunk 1]
```diff
@@ -395,10 +291,9 @@ clean:
 	$(DEL) *.obj *.lib *.map
 	$(DEL) msgs.h msgs.c
 	$(DEL) elxxx.c cdxxx.c optab.c debtab.c fltables.c tytab.c
-	$(DEL) impcnvtab.d impcnvtab.c impcnvgen.exe optabgen.exe
 	$(DEL) id.h id.c id.d
 	$(DEL) verstr.h
-	$(DEL) $(MAGICPORT) $(MAGICPORTDIR)\*.obj
+	$(DEL) impcnvtab.c impcnvgen.exe optabgen.exe
```
Comment (yebblies) [inline src/win32.mak]: impcnvtab.d?

**Comment by yebblies** [comment]
Needs https://github.com/D-Programming-Language/dmd/pull/4927

**Comment by MartinNowak** [comment]
updated

**Comment by yebblies** [comment]
Auto-merge toggled on

**Comment by MartinNowak** [comment]
thx

**Comment by Temtaime** [comment]
DDMD is not buildable with msvc due name mangling.
Anyone to fix it ?

**Comment by rainers** [comment]
> DDMD is not buildable with msvc due name mangling and troubles with longdouble.
> Anyone to fix it ?

I noticed this, too. Not sure if I'll be able to provide patches. See also https://github.com/D-Programming-Language/dmd/pull/4929

**Comment by yebblies** [comment]
Please file bugs for the mangling problems.  I wouldn't be surprised to find some codegen problems too.

**Comment by rainers** [comment]
> Please file bugs for the mangling problems.

https://issues.dlang.org/show_bug.cgi?id=14951

**Comment by CyberShadow** [comment]
This pull request caused a ~30% increase in compilation real time for the "Vibe.d diet templates" test on http://digger.k3.1azy.net/trend/

Maybe because of this?

> drop -O -inline by default (too slow)

**Comment by CyberShadow** [comment]
Currently bisecting why older versions of D stopped building on Arch Linux and bisection shows this PR as the one after which things build successfully. However, 28f91993c2ee8c0e752e778c1d344dcfec469664 mixes a lot of various changes, including changes in behaviour and no-op refactorings. 

Please, make your commits more granular!

**Pull request merged**